core/src/main/java/org/elasticsearch/bootstrap/Spawner.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.bootstrap;
+
+import org.apache.lucene.util.Constants;
+import org.elasticsearch.env.Environment;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Spawns native plugin controller processes if present.  Will only work prior to seccomp being set up.
+ */
+final class Spawner implements Closeable {
+
+    private static final String PROGRAM_NAME = Constants.WINDOWS ? "controller.exe" : "controller";
+    private static final String PLATFORM_NAME = makePlatformName(Constants.OS_NAME, Constants.OS_ARCH);
+    private static final String TMP_ENVVAR = "TMPDIR";
+
+    /**
+     * On Windows we have to retain a reference to the OutputStream that corresponds to each
+     * process's stdin, otherwise the pipe will be closed and the spawned process will receive
+     * an EOF on its stdin.  This isn't necessary on *nix for daemons that only need to shut
+     * down when the JVM exits, but storing these references improves testability by making this
+     * class Closeable.
+     */
+    private final List<OutputStream> stdinReferences = new ArrayList<>();
+
+    @Override
+    public void close() throws IOException {
+        for (OutputStream stream : stdinReferences) {
+            stream.close();
+        }
```

Comment:
lets call `stdinReferences.clear()` after we closed all of them. I also think you should use `IOUtils.close(stdinReferences)` instead, it will close all references even if one close call throws an exception.